passes/techmap/zinit.cc

```diff
@@ -139,11 +142,21 @@ struct ZinitPass : public Pass {
 
 				cell->setPort(ID::D, sig_d);
 				cell->setPort(ID::Q, initwire);
-			}
 
-			for (auto &it : initbits)
-				if (donebits.count(it.first) == 0)
-					log_error("Failed to handle init bit %s = %s.\n", log_signal(it.first), log_signal(it.second));
+				if (cell->type == ID($adff)) {
+					auto val = cell->getParam(ID::ARST_VALUE);
+					for (auto &b : val)
```

Comment:
You should only invert bits that actually correspond to State::S1 in initval.  Also note that this doesn't consider the State::Sx possibility in the reset value.